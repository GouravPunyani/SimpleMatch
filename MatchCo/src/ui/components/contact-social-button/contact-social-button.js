// @flow

import React from 'react'
import { StyleSheet, Image, TouchableOpacity } from 'react-native'

export type Props = {
  network: 'facebook' | 'instagram' | 'twitter',
  onPress: () => *
};

const icons = {
  facebook: require('./facebook.png'),
  instagram: require('./instagram.png'),
  twitter: require('./twitter.png')
}

export function ContactSocialButton (props: Props): React.Children {
  return (
    <TouchableOpacity style={styles.wrapper} onPress={props.onPress}>
      <Image
        style={styles.icon}
        source={icons[props.network]}
        resizeMode='contain'
      />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  wrapper: {},
  icon: { width: 38, height: 38 }
})
