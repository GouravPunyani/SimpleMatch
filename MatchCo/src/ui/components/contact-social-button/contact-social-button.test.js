// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ContactSocialButton as Comp } from './contact-social-button'

test('renders correctly', () => {
  const tree = renderer.create(
    <Comp network='facebook' onPress={() => null} />
  )
  expect(tree).toMatchSnapshot()
})
