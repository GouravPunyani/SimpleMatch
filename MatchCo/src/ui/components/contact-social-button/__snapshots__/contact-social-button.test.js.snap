// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 38,
        "width": 38,
      }
    }
  />
</View>
`;
